clap_generate/src/man.rs

```diff
@@ -0,0 +1,287 @@
+use std::fmt::write;
+
+use clap::ArgSettings;
+
+/// Man page generator
+#[derive(Debug, Clone)]
+pub struct Man<'a> {
+    app: clap::App<'a>,
+    section: i8,
+    manual: Option<String>,
+    sections: Vec<(String, Vec<String>)>,
+}
+
+impl<'a> Man<'a> {
+    /// Create a new man page for the app.
+    pub fn create(
+        app: clap::App<'a>,
+        section: i8,
+        manual: Option<String>,
+        buf: &mut dyn std::io::Write,
+    ) {
+        let man = Man {
+            app,
+            section,
+            manual,
+            sections: Vec::new(),
+        };
+
+        man.write(buf)
+    }
+
+    /// Create a new builder for man pages.
+    pub fn new(app: clap::App<'a>, section: i8, manual: Option<String>) -> Self {
```

Comment:
We should document what peo0ple should set section and manual to.

If there are sane defaults, we should probably use those with builder methods rather than requiring them in `new`

Reply:
I added some comments about this and converted `Man` to use the builder-pattern.